### PR TITLE
Various wallet fixes, accessors, and observer events

### PR DESCRIPTION
**Pull request by bbondy**
Covered in this PR:

- Adds the ability for mojo clients to get/set the auto lock timeout in minutes
- Add observer events for auto lock minutes, selected account changed, and default wallet changed
- Fixes a settings page bug with auto lock
- Adds tests for all of this and similar things

Resolves https://github.com/brave/brave-browser/issues/18438
Resolves https://github.com/brave/brave-browser/issues/18508
Resolves https://github.com/brave/brave-browser/issues/17786
Resolves https://github.com/brave/brave-browser/issues/18366
Resolves https://github.com/brave/brave-browser/issues/18557

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/ui/webui/settings/brave_wallet_handler.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+ private:
+  // SettingsPageUIHandler overrides:
+  void RegisterMessages() override;
+  void OnJavascriptAllowed() override {}
+  void OnJavascriptDisallowed() override {}
+
+  void GetAutoLockMinutes(base::Value::ConstListView args);
+
+  DISALLOW_COPY_AND_ASSIGN(BraveWalletHandler);
```
Comment (yrliou) [inline browser/ui/webui/settings/brave_wallet_handler.h:26]: This is deprecated, we should stop using it in new classes.

Comment (bbondy) [reply browser/ui/webui/settings/brave_wallet_handler.h:26]: Oops copied from existing.

**Comment by bbondy** [comment]
not ready for re-review yet, I'll ping. Some rebase problem.

File: browser/brave_wallet/keyring_controller_unittest.cc [hunk 2]
```diff
@@ -1735,9 +1759,20 @@ TEST_F(KeyringControllerUnitTest, SetSelectedAccount) {
 
 TEST_F(KeyringControllerUnitTest, GetSetAutoLockMinutes) {
```
[the remark this answers is not in the document]
Comment (bbondy) [reply browser/brave_wallet/keyring_controller_unittest.cc:1899]: Fixed in https://github.com/brave/brave-core/pull/10392/commits/048f7e5adff9fa945c6c965c0a1856518e863062

File: browser/ui/webui/settings/brave_wallet_handler.h (hunk 1 above)
Comment (bbondy) [reply browser/ui/webui/settings/brave_wallet_handler.h]: rebased out here https://github.com/brave/brave-core/pull/10392/files#diff-76a0cf7b9026674ba12a29e20a1817d42fddf0b51d9f8496d6331c312ed78155R26

**Review by yrliou — APPROVED** [review]
LGTM

**Pull request merged**